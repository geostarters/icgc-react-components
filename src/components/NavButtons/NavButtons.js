//@flow

import React, { Component } from "react";
import { Button as ButtonSUI } from "semantic-ui-react";
import PropTypes from "prop-types";

// eslint-disable-next-line no-unused-vars
import styles from "./NavButtons.css";

export default class NavButtons extends Component {

	render() {

		return (
			<ButtonSUI.Group className={this.props.fullWidth ? "full-width" : ""}>
				<ButtonSUI basic color={this.props.color ? this.props.color : "yellow"} disabled={this.props.disabledPrev} floated='left' labelPosition='left' icon='left chevron' content='Anterior' onClick={this.props.goPrevStep}/>
				<ButtonSUI color={this.props.color ? this.props.color : "yellow"} disabled={this.props.disabledNext} floated='right' labelPosition='right' icon='right chevron' content='SegÃ¼ent' onClick={this.props.goNextStep}/>
			</ButtonSUI.Group>
		);

	}

}

NavButtons.propTypes = {
	goPrevStep: PropTypes.func,
	goNextStep: PropTypes.func,
	disabledPrev: PropTypes.bool,
	disabledNext: PropTypes.bool,
	color: PropTypes.string,
	fullWidth: PropTypes.bool
};
