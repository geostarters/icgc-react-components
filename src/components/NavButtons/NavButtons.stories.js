//@flow

import React from "react";

import { action } from "@storybook/addon-actions";

import NavButtons from "./NavButtons";

import "semantic-ui-css/semantic.min.css";

export default {
	title: "NavButtons",
	component: NavButtons
};

const Template = args => (<NavButtons {...args}/>);

export const AnteriorISeguent = Template.bind({});
AnteriorISeguent.storyName = "Anterior i Seguent";
AnteriorISeguent.args = {
	goPrevStep: action("goPrevStep"),
	goNextStep: action("goNextStep")
};

export const AmbAmplada100 = Template.bind({});
AmbAmplada100.storyName = "Amb amplada 100%";
AmbAmplada100.args = {
	goPrevStep: action("goPrevStep"),
	goNextStep: action("goNextStep"),
	color: "teal",
	fullWidth: true,
	prevText: "Prev",
	nextText: "Next"
};

export const AmbAmplada100NextDisable = Template.bind({});
AmbAmplada100NextDisable.storyName = "Amb amplada 100% next disable";
AmbAmplada100NextDisable.args = {
	goPrevStep: action("goPrevStep"),
	goNextStep: action("goNextStep"),
	color: "teal",
	fullWidth: true,
	prevText: "Prev",
	nextText: "Next",
	disabledNext: true
};
