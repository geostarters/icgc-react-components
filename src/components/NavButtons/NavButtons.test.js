//@flow

import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import React from "react";
import NavButtons from "./NavButtons";
import { Button as ButtonSUI } from "semantic-ui-react";

describe("NavButtons", () => {

	it("match the snapshot ", () => {

		const mockProps = {
			color: "teal"
		};

		const component = shallow(<NavButtons {...mockProps}></NavButtons>);
		expect(toJSON(component)).toMatchSnapshot();

	});


	it("to have two buttons ", () => {

		const component = shallow(<NavButtons></NavButtons>);
		expect(component.find(ButtonSUI)).toHaveLength(2);

	});

	it("has prop named color = teal ", () => {

		const mockProps = {
			color: "teal"
		};

		const component = shallow(<NavButtons {...mockProps}></NavButtons>);
		const buttons = component.find(ButtonSUI);
		const button = buttons.at(0);
		expect("color" in button.props()).toEqual(true);
		expect(button.props().color).toEqual("teal");

	});

	it("it has a goPrevStep function that is called when click ", () => {

		const mockProps = {
			goPrevStep: jest.fn(),
			goNextStep: jest.fn()
		};

		const component = shallow(<NavButtons {...mockProps}></NavButtons>);
		const buttons = component.find(ButtonSUI);
		buttons.at(0).simulate("click");
		expect(mockProps.goPrevStep).toBeCalled();

	});


});
