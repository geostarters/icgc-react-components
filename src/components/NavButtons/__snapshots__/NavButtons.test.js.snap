// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavButtons match the snapshot  1`] = `
<ButtonGroup
  className=""
>
  <Button
    as="button"
    basic={true}
    color="teal"
    content="Anterior"
    floated="left"
    icon="left chevron"
    labelPosition="left"
  />
  <Button
    as="button"
    color="teal"
    content="SegÃ¼ent"
    floated="right"
    icon="right chevron"
    labelPosition="right"
  />
</ButtonGroup>
`;
